src/sentry/notifications/notifications/strategies/role_based_recipient_strategy.py

```diff
@@ -43,23 +63,43 @@ def determine_recipients(
         # convert members to users
         return user_service.get_many(filter={"user_ids": [member.user_id for member in members]})
 
-    @abstractmethod
     def determine_member_recipients(self) -> Iterable[OrganizationMember]:
         """
         Depending on the type of request this might be all organization owners,
         a specific person, or something in between.
         """
-        raise NotImplementedError
+        members: Iterable[
+            OrganizationMember
+        ] = OrganizationMember.objects.get_contactable_members_for_org(self.organization.id)
+
+        if not self.scope and not self.role:
+            return members
+
+        valid_roles = []
+        if self.scope:
+            valid_roles.extend([r.id for r in roles.get_all() if r.has_scope(self.scope)])
+        if self.role:
+            valid_roles.extend([self.role.id])
+
+        member_ids = self.organization.get_members_with_org_roles(roles=valid_roles).values_list(
+            "id", flat=True
+        )
+        # ignore type because of optional filtering
+        members = members.filter(id__in=member_ids)  # type: ignore[attr-defined]
+
+        if self.scope:
+            for member in members:
```

Comment:
@cathyteng17 Nit: I would add an assertion here to make sure scope and role are not both set at the same time